aten/src/ATen/native/xnnpack/Add.cpp

```diff
@@ -0,0 +1,228 @@
+#ifdef USE_XNNPACK
+
+#include <ATen/native/xnnpack/Allocator.h>
+#include <ATen/native/xnnpack/Common.h>
+
+namespace at {
+namespace native {
+namespace xnnpack {
+namespace internal {
+namespace add {
+
+struct Context final {
+  Operator add_op;
+
+  static constexpr float kMin = -std::numeric_limits<float>::infinity();
+  static constexpr float kMax = std::numeric_limits<float>::infinity();
+};
+
+namespace {
+
+// Supports NHWC and NCHW FP32 adds.
+
+bool available(
```

Comment:
What's the difference between `available` and `usable` functions? Both seem to be runtime checks on the sizes of the inputs.

Reply:
Removing Add.